### PR TITLE
Adding Heroku

### PR DESCRIPTION
**Pull request by jyee**
### What does this PR do?
This PR adds the Heroku buildpack to the Basic Agent Usage section.

### Motivation
I think that the Heroku Buildpack most accurately fits into the Agent section, since it basically installs the Agent onto a Heroku dyno. It's not really an integration in the typical sense. But I'd love to discuss if someone thinks it should be listed in the Integrations instead.

### Preview link
https://docs-staging.datadoghq.com/jyee/heroku_integration/agent/basic_agent_usage/heroku/

### CODE REVIEW

**Comment by jyee** [comment]
Outstanding tasks:
- A few minor text fixes
- Image needs to be shrunk a bit, the logo is a bit big compared to the others
- Need to add item to the navigation menu

File: content/agent/basic_agent_usage/heroku.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+title: Heroku Buildpack
+kind: documentation
+---
+
+This buildpack installs the Datadog Agent in your Heroku Dyno, allowing you to collect system metrics, custom application metrics and traces. To collect custom application metrics or traces, you will also need to include the language appropriate [DogStatsD or Datadog APM library](http://docs.datadoghq.com/libraries/).
```
Comment (l0k0ms) [inline content/agent/basic_agent_usage/heroku.md]: r/`,you will also need to include the`/`include the`

File: content/agent/basic_agent_usage/heroku.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+The Datadog agent provides a listening port on 8125 for statsd/dogstatsd metrics and events. Traces are collected on port 8126.
+
+## Configuration
+
+In addition to the environment variables shown above, there are a number of others you can set:
+
+| Setting | Description|
+| --- | --- |
+| DD_API_KEY | *Required.* Your API key is available from the [Datadog API integrations](https://app.datadoghq.com/account/settings#api) page. Note that this is the *API* key, not the application key. |
```
Comment (l0k0ms) [inline content/agent/basic_agent_usage/heroku.md]: variables should be in code blocks i.e: `DD_API_KEY`

File: content/agent/basic_agent_usage/heroku.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+| DD_SERVICE_NAME | *Optional.* While not read directly by the Datadog Agent, we highly recommend that you set an environment variable for your service name. See the [Service Name](#service-name) section below for more information. |
+| DD_SERVICE_ENV | *Optional.* The Datadog Agent will automatically try to identify your environment by searching for a tag in the form `env:<your environment name>`. If you do not set a tag or wish to override an existing tag, you can set the environment with this setting. For more information, see the [Datadog Tracing environments page](https://docs.datadoghq.com/tracing/environments/). |
+
+### Histogram percentiles
+
+You can optionally set additional percentiles for your histogram metrics. By default only the 95th percentile will be generated. To generate additional percentiles, set *all* percentiles, including the default, using the env variable `DD_HISTOGRAM_PERCENTILES`.  For example, if you want to generate 0.95 and 0.99 percentiles, you may use following command:
```
Comment (l0k0ms) [inline content/agent/basic_agent_usage/heroku.md]: r/`You can optionally`/`Optionally,`

Comment (l0k0ms) [inline content/agent/basic_agent_usage/heroku.md]: r/`By default only the 95th percentile will be generated.`/`By default only the 95th percentile is generated.`

Comment (l0k0ms) [inline content/agent/basic_agent_usage/heroku.md]: r/`you may use following command:`/`use the following command:`

File: content/agent/basic_agent_usage/heroku.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+Although the service name is passed to Datadog on the application level, we highly recommend that you set the value as an environment variable, rather than directly in your application code.
+
+For example, set your service name as an environment variable:
+
+```shell
+heroku config:set DD_SERVICE_NAME=my-webapp
+```
+
+Then in a python web application, you could set the service name from the environment variable:
```
Comment (l0k0ms) [inline content/agent/basic_agent_usage/heroku.md]: r/`you could set the service name`/`set the service name`

**Comment by jyee** [comment]
I think this is ready to go. The table looks horrible because word-break styling that should get fixed as part of #2280

**Comment by jeremy-lq** [comment]
https://github.com/DataDog/documentation/pull/2284 has been merged to address the issue.

**Pull request merged**